### PR TITLE
Preferences Fragment

### PR DESCRIPTION
**Pull request by ricdezen**
Very very tiny pull request containing a Fragment that can be used to display the shared preferences (mainly the password). I had to touch up just a couple of lines in the Password manager to make this compatible. Sadly I could not prepare a test, I just spent 2 hours trying but I believe there may be a glitch with PreferenceFragmentCompat

### CODE REVIEW

File: app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Fragment dedicated to showing the Preferences for the app.
+ *
+ * @author Riccardo De Zen
+ */
+public class SettingsFragment extends PreferenceFragmentCompat {
+
+    /**
+     * Default constructor.
+     */
+    public SettingsFragment() {
```
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057)

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/androidTest/java/ingsw/group1/findmyphone/AndroidTestUtils.java  1
- app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057&bid=16566722)

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/androidTest/java/ingsw/group1/findmyphone/AndroidTestUtils.java  1
- app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057&bid=16566722)

File: app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java (hunk 1 above)
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057)

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/AndroidTestUtils.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057&bid=16566722)

File: app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    /**
+     * Method called to inflate the Preferences hierarchy.
+     *
+     * @param savedInstanceState The saved instance state.
+     * @param rootKey            The root key, for the hierarchy, it is provided automatically.
+     */
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.app_preferences, rootKey);
```
Comment (turcato-niccolo) [inline app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: The root key will the password for encrypting outgoing messages? (I've fallen a little behind the logic)

Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: Not really important in this case, a `PreferenceScreen` in a preferences xml file can have a certain key, if a certain screen key is provided to this method then the fragment should use it as root for the preferences to show. I think when called by the system will be null by default meaning to use the default screen in a file, which is our case. I wanted to put it always null but it didn't seem to hurt like this

Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: Also this should call the superclass method not sure why I have forgot

Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: image

Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: Ah yes I see you got my not so subtle message

Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: Gotta try that game someday

Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: `honk`

Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java:24]: > Not really important in this case, a `PreferenceScreen` in a preferences XML file can have a certain key, if a certain screen key is provided to this method then the fragment should use it as root for the preferences to show. I think when called by the system will be null by default meaning to use the default screen in a file, which is our case. I wanted to put it always null but it didn't seem to hurt like this

Now, being serious, I kinda get it 👌

**Review by bortoletti-giorgia — DISMISSED** [review]
I approve this pull request.
In the future we have to add a "guide" with the app's instructions.

**Review by kummy98 — DISMISSED** [review]

**Comment by ricdezen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/androidTest/java/ingsw/group1/findmyphone/AndroidTestUtils.java  1
- app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5027057&bid=16566722)

File: .idea/misc.xml [hunk 3]
```diff
@@ -9,7 +9,7 @@
     <option name="OUTPUT_DIRECTORY" value="$PROJECT_DIR$/javadoc" />
     <option name="OPTION_SCOPE" value="private" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (ricdezen) [inline .idea/misc.xml:12]: But fookin why tho

Comment (turcato-niccolo) [reply .idea/misc.xml:12]: Senza titolo (2)

**Pull request merged**